packages/sycamore-web/src/lib.rs

```diff
@@ -35,6 +35,40 @@ pub trait Html: GenericNode<EventType = Event, PropertyType = JsValue> {
     const IS_BROWSER: bool;
 }
 
+/// Create a `GenericNode` instance from a `web_sys::HtmlElement`.
+///
+/// When the `GenericNode` is a `DomNode` or a `HydrateNode` then the
+/// web_sys::Element will be reused and won't cause another node to be
+/// created in the browser DOM.
+pub fn from_web_sys<G: GenericNode>(element: web_sys::Element) -> G {
```

Comment:
Also I think the param can just be `web_sys::Node` since this would also work with text nodes and comment nodes. You'll also need to update the doc comment as well.